WalletWasabi.Fluent/Models/Wallets/WalletTransactionsModel.cs

```diff
@@ -23,30 +24,31 @@ public partial class WalletTransactionsModel : ReactiveObject
 
 	public WalletTransactionsModel(Wallet wallet)
 	{
+		_wallet = wallet;
+
 		TransactionProcessed =
 			Observable.FromEventPattern<ProcessedResult?>(wallet, nameof(wallet.WalletRelevantTransactionProcessed)).ToSignal()
 					  .Merge(Observable.FromEventPattern(wallet, nameof(wallet.NewFiltersProcessed)).ToSignal())
 					  .Sample(TimeSpan.FromSeconds(1))
 					  .ObserveOn(RxApp.MainThreadScheduler)
 					  .StartWith(Unit.Default);
 
-		List =
+		var transactionChanges =
 			Observable.Defer(() => BuildSummary().ToObservable())
 					  .Concat(TransactionProcessed.SelectMany(_ => BuildSummary()))
 					  .ToObservableChangeSet(x => x.GetHash());
-		_wallet = wallet;
+
+		transactionChanges.Bind(out var collection).Subscribe();
+		Transactions = collection;
 	}
 
-	public IObservable<Unit> TransactionProcessed { get; }
+	public ReadOnlyObservableCollection<TransactionSummary> Transactions { get; }
 
-	public IObservable<IChangeSet<TransactionSummary, uint256>> List { get; }
+	public IObservable<Unit> TransactionProcessed { get; }
 
-	public async Task<TransactionSummary?> GetById(string transactionId)
+	public TransactionSummary? GetById(uint256 transactionId)
```

Comment:
Can we do TryGet pattern to return a not nullable object with [NotNullWhen(true)]? It is a minor change, but aligns with "avoid the nullables as much as possible"